### PR TITLE
ci: enforce lower ci job timeout

### PR DESCRIPTION
**Pull request by mhofman**
Lower the timeout of some all packages CI jobs

### CODE REVIEW

File: .github/workflows/test-all-packages.yml [hunk 1]
```diff
@@ -45,6 +45,7 @@ jobs:
   # We split the package tests into two jobs because type linting
   # is inefficient and slow https://github.com/typescript-eslint/typescript-eslint/issues/2094
   lint-primary:
+    timeout-minutes: 10
```
Comment (turadg) [inline .github/workflows/test-all-packages.yml:45]: looks like this one is usually under 5min but better to have its timeout match the other one

**Review by turadg — APPROVED** [review]
LGTM as a squash PR

**Comment by mhofman** [comment]
@mergifyio update

**Comment by mhofman** [comment]
@mergifyio squash

**Pull request merged**